ddtrace/tracer/sampler.go

```diff
@@ -185,42 +186,54 @@ func newRulesSampler(rules []SamplingRule) *rulesSampler {
 	}
 }
 
-// appliedSamplingRules validates the user-provided rules and returns an internal representation.
-// If the DD_TRACE_SAMPLING_RULES environment variable is set, it will replace the given rules.
-func appliedSamplingRules(rules []SamplingRule) []SamplingRule {
+func samplingRulesFromEnv() ([]SamplingRule, error) {
 	rulesFromEnv := os.Getenv("DD_TRACE_SAMPLING_RULES")
-	if rulesFromEnv != "" {
-		rules = rules[:0]
-		jsonRules := []struct {
-			Service string      `json:"service"`
-			Name    string      `json:"name"`
-			Rate    json.Number `json:"sample_rate"`
-		}{}
-		err := json.Unmarshal([]byte(rulesFromEnv), &jsonRules)
+	if rulesFromEnv == "" {
+		return nil, nil
+	}
+	//var rules []SamplingRule
+	rules := make([]SamplingRule, 0)
+	jsonRules := []struct {
+		Service string      `json:"service"`
+		Name    string      `json:"name"`
+		Rate    json.Number `json:"sample_rate"`
+	}{}
+	err := json.Unmarshal([]byte(rulesFromEnv), &jsonRules)
+	if err != nil {
+		log.Warn("error parsing DD_TRACE_SAMPLING_RULES: %v", err)
+		return nil, fmt.Errorf("error parsing DD_TRACE_SAMPLING_RULES: %v", err)
+	}
+	var errStr string
+	for _, v := range jsonRules {
+		if v.Rate == "" {
+			log.Warn("error parsing rule: rate not provided")
+			errStr += "error parsing rule: rate not provided"
+			continue
+		}
+		rate, err := v.Rate.Float64()
 		if err != nil {
-			log.Warn("error parsing DD_TRACE_SAMPLING_RULES: %v", err)
-			return nil
+			log.Warn("error parsing rule: invalid rate: %v", err)
+			errStr += fmt.Sprintf("error parsing rule: invalid rate: %v", err)
```

Comment:
It seems like these error strings will be merged without a space between them. Since there's no test for them, you've probably missed that. Also, it might be worth not repeating "error parsing rule" in each and every one.